### PR TITLE
"scopeFilter" return wrong data in getPattern APIs #545

### PR DESCRIPTION
**Pull request by chaoq0318**
fix issue:
   "scopeFilter" return wrong data in getPattern APIs #545
   Confuse description for scopeFilter in getPattern APIs #546

### CODE REVIEW

File: g11n-ws/vip-common/src/main/java/com/vmware/vip/common/constants/ConstantsMsg.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
    public static final String PARAM_NOT_VALIDATE = "The parameter is not pass the validate";
    public static final String PATTERN_NOT_VALIDATE = "The request pattern name is invalid";
    public static final String FIFE_NOT_FOUND= "File is not existing";
+   public static final String SCOPE_FILTER_NOT_VALIDATE = "The request parameter scope filter is invalid, only one filtering method is supported";
```
Comment (huihuiw01) [inline g11n-ws/vip-common/src/main/java/com/vmware/vip/common/constants/ConstantsMsg.java:30]: scope filter should be 'scopeFilter'

File: g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/pattern/PatternServiceImpl.java [hunk 2]
```diff
@@ -240,10 +240,23 @@ private void filterScope(Map<String, Object> patternMap, String scopeFilter) {
 			return;
 		}
 
-		Arrays.asList(scopeFilter.split(ConstantsChar.COMMA)).stream().forEach(scopeNode -> {
-			List<String> scopeFilters = Arrays.asList(scopeNode.split(ConstantsChar.UNDERLINE));
-			removeData(patternMap, scopeFilters, 0);
-		});
+		// If scopeFilter starts with ^, it means deleting the data under the node
```
[the remark this answers is not in the document]
Comment (chaoq0318) [reply g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/pattern/PatternServiceImpl.java:243]: In the buildPatternMap method, after obtaining the data of categories, there are other processing logics in the follow-up, so it is not recommended to put in the getCategories method.

File: g11n-ws/vip-common/src/main/java/com/vmware/vip/common/constants/ConstantsMsg.java (hunk 1 above)
Comment (chaoq0318) [reply g11n-ws/vip-common/src/main/java/com/vmware/vip/common/constants/ConstantsMsg.java]: fixed

File: g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/pattern/PatternServiceImpl.java [hunk 3]
```diff
@@ -240,10 +240,23 @@ private void filterScope(Map<String, Object> patternMap, String scopeFilter) {
...
+		if (scopeFilter.startsWith(ConstantsChar.REVERSE)) {
```
[the remark this answers is not in the document]
Comment (chaoq0318) [reply g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/pattern/PatternServiceImpl.java:244]: Currently only supports forward or reverse, this format is temporarily not supported.

**Pull request merged**